esi_leap/api/controllers/v1/node.py

```diff
@@ -60,14 +62,38 @@ def __init__(self, **kwargs):
 
 class NodesController(rest.RestController):
 
+    def get_all_nodes(self):
+        nodes = ironic.get_node_list(context)
+        return nodes
+
+    def get_project_list(self):
+        plist = keystone.get_project_list()
+        return plist
+
     @wsme_pecan.wsexpose(NodeCollection)
     def get_all(self):
         context = pecan.request.context
-        nodes = ironic.get_node_list(context)
+        # t1 = threading.Thread(target=get_all_nodes)
+
+        nodes = None
+        project_list = None
+
+        with concurrent.futures.ThreadPoolExecutor() as executor:
+            f1 = executor.submit(get_all_nodes)
```

Comment:
I think you can avoid having a new function and just have

```
f1 = executor.submit(ironic.get_node_list, context)
```

Reply:
Okay, will try!
Also, I am running unit tests but not sure how I can test time